crates/pico_macros/src/source.rs

```diff
@@ -0,0 +1,58 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+pub(crate) fn source(_args: TokenStream, item: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(item as DeriveInput);
+
+    let struct_name = input.clone().ident;
+    let key = struct_name.to_string();
+
+    let output = quote! {
+        #[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
I think it's common practice to have the user derive Debug, Clone, etc. themselves, in case they require a custom impl. Wdyt?